mobile-widgets/qmlmanager.cpp

```diff
@@ -518,6 +541,8 @@ void QMLManager::finishSetup()
 		appendTextToLog(tr("no cloud credentials"));
 		setStartPageText(RED_FONT + tr("Please enter valid cloud credentials.") + END_FONT);
 	}
+	m_initialized = true;
+	emit initializedChanged();
```

Comment:
Does Qt really not provide a setProperty macro that sets the property and sends the signal? That feels so lame!

Reply:
The documentation EXPLICITLY states that if you use `MEMBER` that setting the variable will emit the signal. Except that in my tests it doesn't.
So yes, I can implement this in the beautifully elegant way envisioned by Qt and simply accept the fact that often the QML UI doesn't get notified... that didn't seem as useful to me as it may sound at first.
Or maybe my testing is broken and this all works in reality. Who knows.
The amount of ugly hacks that we have all over the place because Qt/QML doesn't do what realistically one thinks it should do. Like rows/columns/roles when it comes to ListViews.
But I'm happy to spend more time on trying to coax it into doing this differently... or simply just writing my own setter function. Which seems so stupid as there is exactly ONE TIME when I change that value from false to true.
Maybe that whole thing should be rethought and instead I should do one of those bizarre calls into QML instead of abusing yet another `bool` value to notify QML of a change.

Yet another thing where our code is chock full of ugly hacks that were added one at a time to fix one issue at one point in time.

Reply:
OK, tested this again. If I remove the emit, the UI never receives the update.

Which means this line `Qt emits automatically that signal when needed for MEMBER properties that do not have an explicit setter.` on this page https://doc.qt.io/qt-5/properties.html is flatly incorrect.

Would you prefer that I implement a setter in order to hide this ugliness from the `finishSetup()` function?

Reply:
> The documentation EXPLICITLY states that if you use MEMBER that setting the variable will emit the signal. 

It can't when you do a simple assignment. It would have to somehow override `operator=()` for the variable. However, since the variable is simply a bool, this is not possible.

So with `setting the property` they surely mean a special function or macro. That's what I was asking for.

There actually is the `setProperty` function. But that is dynamic, so it is now very appealing.

> Maybe that whole thing should be rethought and instead I should do one of those bizarre calls into QML instead of abusing yet another bool value to notify QML of a change.

I think the Q_PROPERTY thing is the way it is supposed to work. As weird as it may be...

Reply:
> OK, tested this again. If I remove the emit, the UI never receives the update.

I know - it can't auto-generate that emit. :)

> Would you prefer that I implement a setter in order to hide this ugliness from the finishSetup() function?

Not really. Even more lines of code. But if you feel it is "cleaner", "clearer" or something, it wouldn't bother me either.